### PR TITLE
geo/geomfn: implement ST_AsBinary({geometry,text})

### LINKED ISSUE #48379: geo/geogfn: implement ST_AsBinary({geography,text})
Implement `ST_AsBinary` on arguments {geography,text}, which should adopt [PostGIS behaviour](https://postgis.net/docs/ST_AsBinary.html).

For Geography builtins, please do the following:
* Ideally add a relevant helper function in [`pkg/geo/geogfn`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo/geogfn) (parse and output related functions can go in [`pkg/geo`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo)). Add exhaustive unit tests here - you can run through example test cases and make sure that PostGIS and CRDB return the same result within a degree of accuracy (1cm for geography).
* Create a new builtin that references this function in [`pkg/sql/sem/builtins/geo_builtins.go`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sem/builtins/geo_builtins.go). Note that we currently do not support optional arguments, so we define functions that have optional arguments once without the optional argument (using the default value in the optional position), and once with the optional argument.
* Modify the tests in [`pkg/sql/logictest/testdata/logic_test/geospatial`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/logictest/testdata/logic_test/geospatial) to call this functionality at least once. You can call `make testbaselogic FILES='geospatial' TESTFLAGS='-rewrite'` to regenerate the output. Tests here should just ensure the builtin is linked end to end.
* Ensure the documentation is regenerated by calling `make buildshort`.
* Submit your PR - make sure to follow the guidelines from [creating your first PR](https://wiki.crdb.io/wiki/spaces/CRDB/pages/181633464/Your+first+CockroachDB+PR]).

You can follow #48529 for an example PR.

<sub>:robot: This issue was synced with a spreadsheet by [gsheets-to-github-issues](https://github.com/cockroachlabs/gsheet-to-github-issues) by [otan](https://github.com/otan) on 2020-05-17T19:28:26Z. Changes to titles, body and labels may be overwritten.</sub>

### PR DESCRIPTION
**Pull request by hueypark**
Fixes #48379, #48870

Release note (sql change): Implemented the geometry based builtins `ST_AsBinary` with encoding.

### CODE REVIEW

**Comment by hueypark** [comment]
@otan I think we also need 'st_ashexwkb' which receives encoding as an additional variable.

**Comment by otan** [comment]
Yep, there is an issue (#48880) for that.

This LGTM!

bors r+

**Comment by otan** [comment]
looks like ./pkg/geo doesn't compile under test.

File: pkg/sql/sem/builtins/geo_builtins.go [hunk 1]
```diff
@@ -560,13 +560,51 @@ var geoBuiltins = map[string]builtinDefinition{
...
+			Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				g := args[0].(*tree.DGeography)
+				text := string(tree.MustBeDString(args[1]))
+
+				wkb, err := geo.EWKBToWKB(g.Geography.EWKB(), geo.StringToByteOrder(text))
+				return tree.NewDBytes(tree.DBytes(wkb)), err
+			},
+			Info: infoBuilder{
+				info: "Returns the WKB representation of a given Geography. " +
+					"The second variant takes a second argument denoting the encoding - using little-endian ('NDR') or big-endian ('XDR') encoding.",
```
Comment (otan) [inline pkg/sql/sem/builtins/geo_builtins.go:604]: can you make this but just "This variant has a second argument denoting the encoding"

Comment (hueypark) [reply pkg/sql/sem/builtins/geo_builtins.go]: Done.

**Comment by hueypark** [comment]
> Yep, there is an issue (#48880) for that.

#48880 is for `st_ashexewkb`. We need `st_ashexwkb`(without e).

> looks like ./pkg/geo doesn't compile under test.

It's corrected.

**Comment by otan** [comment]
Ah. Yeah that's reasonable - PostGIS doesn't seem to have it but I don't
thonk it hurts to include it.

**Comment by hueypark** [comment]
@otan Would file an issue for `st_ashexwkb` and merge this? I will fix that later.

**Review by otan — APPROVED** [review]

**Comment by hueypark** [comment]
@otan I looked through PostGIS document and we should not make `st_ashexwkb` with encoding.
If you make it, it breaks compatibility.

FYI https://postgis.net/docs/RT_ST_AsHexWKB.html

**Pull request merged**